src/main/ruby/core/argf.rb

```diff
@@ -575,4 +625,7 @@ def advance!
 # from $stdin if no files were given.) Usable like an IO.
 #
 ARGF = Rubinius::ARGFClass.new(ARGV)
+Truffle::KernelOperations.define_hooked_variable(:$FILENAME, -> { ARGF.filename }, -> _ { raise Error, 'Attempt to set $FILENAME' } )
+Truffle::KernelOperations.define_hooked_variable(
+  :$., -> { ARGF.instance_variable_get(:@last_lineno) }, -> value { ARGF.instance_variable_set(:@last_lineno, value) } )
 $< = ARGF
```

Comment:
Can you keep the `$< = ARGF` just below the assignment of ARGF?
Re style, it seems nicer to have one argument per line if the line is too long.